### PR TITLE
[IMP] purchase_order_product_recommendation: more filters

### PR DESCRIPTION
**Pull request by chienandalu**
Bring changes from https://github.com/OCA/purchase-workflow/pull/805 as well as split tests for better inherability

- Extract domain build to an independent method for easy extension by
third modules
- Extract product filter to an independent method for easy extension.
- New product internal category filter.
- New "Show all products" filter that allows to easily show all
purchable products.
- Make tests extensible by other modules

cc @Tecnativa TT20722

### CODE REVIEW

**Review by pedrobaeza — COMMENTED** [review]
Although I didn't tell you before in the other PR, instead of creating such big diff, it's better to leave both classes in the same file (as they are very related), and previous conventions on the names are for putting `Base` for the base class and the same without base for the normal one.

File: purchase_order_product_recommendation/tests/test_recommendation.py [hunk 1]
```diff
@@ -142,6 +151,12 @@ def wizard(self):
         wizard._generate_recommendations()
         return wizard
 
+
+class RecommendationCaseTests(RecommendationCase):
+    @classmethod
+    def setUpClass(cls):
```
Comment (pedrobaeza) [inline purchase_order_product_recommendation/tests/test_recommendation.py]: No need of this

Comment (chienandalu) [reply purchase_order_product_recommendation/tests/test_recommendation.py]: Yes, sorry

**Review by pedrobaeza — APPROVED** [review]

File: purchase_order_product_recommendation/wizards/purchase_order_recommendation.py [hunk 2]
```diff
@@ -43,10 +43,20 @@ class PurchaseOrderRecommendation(models.TransientModel):
...
     )
     show_all_partner_products = fields.Boolean(
-        string='Show all products',
+        string='Show all supplier products',
         default=False,
         help='Show all products with supplier infos for this supplier',
     )
+    show_all_products = fields.Boolean(
+        string='Show all purchasable products',
+        default=False,
```
Comment (nikul-serpentcs) [inline purchase_order_product_recommendation/wizards/purchase_order_recommendation.py:52]: @chienandalu FYI no need to add `default=False`, Boolean always False

**Review by rafaelbn — APPROVED** [review]

**Pull request merged**